core/src/main/java/com/scalar/db/storage/cassandra/Cassandra.java

```diff
@@ -100,19 +98,34 @@ public Optional<Result> get(Get get) throws ExecutionException {
     get = copyAndSetTargetToIfNot(get);
     operationChecker.check(get);
 
-    ResultSet resultSet = handlers.select().handle(get);
-    Row row = resultSet.one();
-    if (row == null) {
-      return Optional.empty();
-    }
-    Row next = resultSet.one();
-    if (next != null) {
-      throw new IllegalArgumentException(
-          CoreError.GET_OPERATION_USED_FOR_NON_EXACT_MATCH_SELECTION.buildMessage(get));
+    Scanner scanner = null;
+    try {
+      if (get.getConjunctions().isEmpty()) {
+        scanner = getInternal(get);
+      } else {
+        scanner = new FilterableScanner(get, getInternal(copyAndPrepareForDynamicFiltering(get)));
```

Comment:
Actually since we already copied the get object here (https://github.com/scalar-labs/scalardb/pull/1682/files#diff-660ca76c9b64d94000e0854e2a18fdc5e6a69c028a74b8cacfa4ef733d4604c2L100), I don't think we need to copy the get object here with `copyAndPrepareForDynamicFiltering()` again.

Reply:
Thank you for the comment. I wanted to pass the original `get` for `FilterableScanner` and the different modified `get` for `getInternal()` since we need the original limit value and projections in `FilterableScanner`. It might be possible to avoid copying the whole `get` operation somehow, but I think the current one is not so bad from the readability perspective. What do you think?

Reply:
That makes sense. Thanks.